src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -536,7 +536,7 @@ public LocalWithInitializer(
                 Debug.Assert(initializer != null);
 
                 _initializer = initializer;
-                _initializerBinder = initializerBinder;
+                _initializerBinder = initializerBinder.GetBinder(initializer);
```

Comment:
>initializerBinder.GetBinder(initializer) [](http://example.com/codeflow?start=37&length=40)

It looks like this operation can return null when  ```LocateDeclaredVariableSymbol``` is doing an error recovery. In that case we might want to inject the binder here explicitly.